dfusion_rust_core/src/models/util.rs

```diff
@@ -220,3 +221,7 @@ pub fn read_amount(bytes: &[u8; 12]) -> u128 {
 
     BigEndian::read_u128(bytes.as_slice())
 }
+
+pub fn map_from_list(arr: &[(u16, u128)]) -> HashMap<u16, u128> {
```

Comment:
ubernit: use `slice` in the function name instead of `list` as you are mapping from a slice and not a list. Also, I would suggest that the function name reflects that this is not for any map, but for price map only, so something like `prices_from_slice` would be better IMO.

If you want to get fancy, you can define a generic `hash_map!` macro that works like the `vec!` one. It would be a great Rust exercise ;)

Reply:
Great suggestion. Would the util file still be the right location to define such a marco?

Reply:
It would be. May I suggest the following syntax (although, I suggest leaving this for separate PR):
```
let map: HashMap<String, i32> = hash_map! {
  "foo": 32,
  "bar": 44,
};
let prices: HashMap<u16, u128> = hash_map! {
    0: 1 * BASE_AMOUNT,
    1: 2 * BASE_AMOUNT,
};
```